examples/lb3-application/lb3app/test/acceptance.js

```diff
@@ -0,0 +1,132 @@
+    .set('Accept', 'application/json')
+    .expect('Content-Type', /json/);
+}
+
+describe('LoopBack 3 style acceptance tests', function () {
+  before(async function () {
+    app = new ExpressServer();
+    await app.boot();
+    await app.start();
+  });
+
+  after(async () => {
+    await app.stop();
+  });
+
+  context('basic REST calls for LoopBack 3 application', () => {
+    it('creates and finds a CoffeeShop', function (done) {
+      json('post', '/api/CoffeeShops')
+        .send({
+          name: 'Coffee Shop',
+          city: 'Toronto',
+        })
+        .expect(200)
+        .end(function (err, res) {
+          assert(typeof res.body === 'object');
+          assert(res.body.name);
+          assert(res.body.city);
+          assert.equal(res.body.name, 'Coffee Shop');
+          assert.equal(res.body.city, 'Toronto');
+          done();
+        });
+    });
+
+    it("gets the CoffeeShop's status", function (done) {
+      json('get', '/api/CoffeeShops/status').expect(200, function (err, res) {
+        res.body.status.should.be.equalOneOf(
+          'We are open for business.',
+          'Sorry, we are closed. Open daily from 6am to 8pm.',
+        );
+        done();
+      });
+    });
+
+    it('gets external route in application', function (done) {
+      json('get', '/ping').expect(200, function (err, res) {
+        assert.equal(res.text, 'pong');
+        done();
+      });
+    });
+  });
+
+  context('authentication', () => {
+    let User;
+
+    before(() => {
+      User = lb3App.models.User;
+    });
+
+    it('creates a User and logs them in and out', function (done) {
+      User.create({email: 'new@email.com', password: 'L00pBack!'}, function (
+        err,
+        user,
+      ) {
+        assert.equal(user.email, 'new@email.com');
+        User.login(
+          {
+            email: 'new@email.com',
+            password: 'L00pBack!',
+          },
+          function (err2, token) {
+            token.should.have.properties('ttl', 'userId', 'created', 'id');
+            assert.equal(token.userId, user.id);
+            User.logout(token.id);
+            User.deleteById(user.id);
+          },
+        );
+        done();
+      });
+    });
+
+    it('rejects anonymous requests to protected endpoints', function (done) {
+      json('get', '/api/CoffeeShops/greet').expect(401, function (err, res) {
+        assert.equal(res.body.error.code, 'AUTHORIZATION_REQUIRED');
+      });
+      done();
+    });
+
+    it('makes an authenticated request', function (done) {
+      User.create({email: 'new@email.com', password: 'L00pBack!'}, function (
+        err,
+        user,
+      ) {
+        user.email.should.be.equal('new@email.com');
+        json('post', '/api/users/login')
+          .send({
+            email: 'new@email.com',
+            password: 'L00pBack!',
+          })
+          .expect(200, function (err2, token) {
+            assert(typeof token.body === 'object');
```

Comment:
When this assertion fails, you will have no idea what was the actual type of the body object. A better solution:

```js
assert.equals(typeof token.body, 'object');
```